### ISSUE TITLE
[Closed] GUI Windows cannot be made Modal.

### ISSUE THREAD

**Issue by Craluminum2413** [ApacheTechSolutions/ApacheTech.VintageMods.CampaignCartographer#7]
**Describe the Bug:**
Scrolling "Colour" dropdown list in "Add New Waypoint Type" window scrolls window behind it ("Pre-Defined Waypoints")

**To Reproduce:**
1. Open "Pre-Defined Waypoints"
2. Press "Add new waypoint type"
3. Press "Colour"
4. In the dropdown menu, hover mouse out of Add New Waypoint Type" area and scroll it
5. It scrolls the window behind it

**Expected Behaviour:**
It scrolls only color dropdown list

**Additional Context:**
![image](https://user-images.githubusercontent.com/69315569/150450872-fb4b952b-21f3-4789-91ac-8023adec391f.png)

**Comment by ApacheTech** [comment]
This is, unfortunately, a bug with the game's UI engine, and not with the mod. There is no way to make a dialogue box Modal, and so they stack on top of each other, and accept input then the mouse in within their bounds.

I've got a lot of work to do with UI in the next update, so I'll have a play around with the engine and see what's possible. Ideally, I'd like an option to show a dialogue box as Modal, so that it has exclusive access to mouse and keyboard inputs, while open. This is something I've started to work towards, with the message boxes, but they're not foolproof yet.

I'll leave this report open, so that people can see it's been addressed, and hopefully a fix can be put in place in the future, as the game's UI engine becomes more robust.
